src/storage/storage.cpp

```diff
@@ -348,46 +345,38 @@ Storage::ReturnCode Storage::Run(int max_wait)
     std::uint64_t number_of_compressed_datasources = 0;
     if (geometry_datasource_input_stream)
     {
-        geometry_datasource_input_stream.read(
-            reinterpret_cast<char *>(&number_of_compressed_datasources),
-            sizeof(number_of_compressed_datasources));
+        number_of_compressed_datasources =
+            io::readElementCount<std::uint64_t>(geometry_datasource_input_stream);
```

Comment:
Why is this element count 64 bit but the other ones above 32 bit? This is so easy to get wrong. In case of a mismatch we probably will crash and burn hard.